### PR TITLE
CAMEL-15581: Fix loading of XML rest definitions with camel-main

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: core/camel-main/src/main/java/org/apache/camel/main/DefaultRoutesCollector.java [hunk 1]
```diff
@@ -206,7 +206,7 @@ public void configure() throws Exception {
                 Set<InputStream> set = resolver.findResources(part);
                 for (InputStream is : set) {
                     log.debug("Found XML rest from location: {}", part);
-                    RestsDefinition rests = (RestsDefinition) ecc.getXMLRoutesDefinitionLoader().loadRoutesDefinition(ecc, is);
+                    RestsDefinition rests = (RestsDefinition) ecc.getXMLRoutesDefinitionLoader().loadRestsDefinition(ecc, is);
```
Comment (davsclaus) [inline core/camel-main/src/main/java/org/apache/camel/main/DefaultRoutesCollector.java:209]: Ha one of the many billion copy/paste mistakes in the world

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
I wonder if we have this problem on 3.4.x branch

**Comment by jamesnetherton** [comment]
> I wonder if we have this problem on 3.4.x branch

Yeah, seems the bug goes back to 3.1.0.

**Comment by ppalaga** [comment]
I dared to revert this because it caused a cycle in the maven dependency graph: https://github.com/apache/camel/pull/4295/checks?check_run_id=1167651159#step:6:15

**Comment by ppalaga** [comment]
The above link shows a different commit now. The error before reverting a95d17c772149a0f3c70243a68532272477c5423 was

```
mvn clean install -DskipTests
[INFO] BuildTimeEventSpy is registered.
[INFO] Scanning for projects...
[ERROR] [ERROR] The projects in the reactor contain a cyclic reference: Edge between 'Vertex{label='org.apache.camel:camel-undertow:3.6.0-SNAPSHOT'}' and 'Vertex{label='org.apache.camel:camel-test-spring-junit5:3.6.0-SNAPSHOT'}' introduces to cycle in the graph org.apache.camel:camel-test-spring-junit5:3.6.0-SNAPSHOT --> org.apache.camel:camel-spring:3.6.0-SNAPSHOT --> org.apache.camel:camel-main:3.6.0-SNAPSHOT --> org.apache.camel:camel-undertow:3.6.0-SNAPSHOT --> org.apache.camel:camel-test-spring-junit5:3.6.0-SNAPSHOT @ 
[ERROR] The projects in the reactor contain a cyclic reference: Edge between 'Vertex{label='org.apache.camel:camel-undertow:3.6.0-SNAPSHOT'}' and 'Vertex{label='org.apache.camel:camel-test-spring-junit5:3.6.0-SNAPSHOT'}' introduces to cycle in the graph org.apache.camel:camel-test-spring-junit5:3.6.0-SNAPSHOT --> org.apache.camel:camel-spring:3.6.0-SNAPSHOT --> org.apache.camel:camel-main:3.6.0-SNAPSHOT --> org.apache.camel:camel-undertow:3.6.0-SNAPSHOT --> org.apache.camel:camel-test-spring-junit5:3.6.0-SNAPSHOT -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectCycleException
```

**Comment by davsclaus** [comment]
Ah yeah you can add such kind of tests in tests/camel-itest - or avoid using undertow in the test, and use a simulated rest component, we do that in some other rest based tests in camel-core testing.

**Pull request merged**